.vscode/launch.json

```diff
@@ -16,7 +16,7 @@
                  "--language",
                  "typescript",
                  "-o",
-                 "C:/Users/vibiret/Desktop/graphtypescriptv4/src",
+                 "C:/sources/gitlocal/graphtypescriptv4/src",
```

Comment:
Is there any way of making this path relative to the repo folder?  This isn't going to be helpful for anyone else.

Reply:
that configuration will only impact vscode users, and no as those are local test repositories (unless we want to commit those test repos here? but I'm afraid it'll get messy). I see this being useful as they already have all the other options configured and only need to update the path.

Reply:
@baywet How about we default to `${workspaceFolder}/output/graphtypescriptv4/src` then add `output/` to the `.gitIgnore`. This will ensure that the output folder is never checked-in and partners get an out-of-the box environment to work with.

Reply:
@peombwa, I am assuming `graphtypescriptv4` is itself a git repo. So I propose the following instead:
`${workspaceFolder}/../graphtypescriptv4` works as a path too in `launch.json`. We can have instructions in the README that `kiota` and `graphtypescriptv4` are checked out at the same root location, which would allow directly outputting the models into the respective repo and every path would still be relative.

In fact, I am working on a similar pattern in the old generator project for `launch.json`. I should get a PR out this week.

Reply:
Here is my proposal: https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/pull/515. Please let me know if similar approach can work for Kiota as well.

Reply:
@peombwa Darrel had originally suggested providing sample projects in the repo itself, but I pushed back on this one. We already have the generator itself + 3x2 projects (abstractions & core). As Kiota matures, abstractions and core will eventually get their own repositories I believe but I want to avoid cramming too many things in this repo.
@zengin thanks for the suggestion, I applied changes among those lines (and took the time to document a lot of aspects that were living in my head until now). let me know what you think.

Reply:
@baywet, I think having generated code accompany these changes would be really useful both for the authors and the reviewers. They are more-or-less free end to end tests. I made a similar comment as @darrelmiller here: https://github.com/microsoft/kiota/pull/60#pullrequestreview-636157702.

If adding that into this repo is too much code crammed into the repo, consider having separate SDK repositories and pointing the diffs in the PRs or using submodules that could be unlinked later.

Thanks for updating the README and the launch.json with relative paths.

Reply:
Would we have those test repos under the Microsoft or Microsoft graph orgs? I'm afraid it'll confuse people with the "production" SDKs. What if I created repositories under my username for the time being?

Reply:
I guess for the time being, it'd be easiest to set it up under your username.

I think it would not be too confusing to keep them under `microsoftgraph` with a clear name and README which shows that it is not meant to be the published SDKs: e.g. `microsoftgraph/kiota-typescript-sample`. But if this becomes the long term solution for testing the evolution of Kiota, it may make sense to have it under `microsoft` too. I am not familiar with the limitations of how many repos we can create under `microsoft` org.

Reply:
Darrel and I had further brainstorming on this and here is a recap.
We have multiple concerns here:
- The launch.json configuration should be valid, the paths should resolve with minimal repos cloning/rework from people cloning this repo.
- The parameters should be documented so people can configure other IDEs, use dotnet run, use a compiled version of kiota... #96
- The unit tests coverage should be ramped up and test aspects, not just generate files #79
- The PRs should provide a generation diff (more on that below)
- We want to avoid overcrowding this repo with too many things

The solution we're suggesting is the following:
- create a samples repo under the Microsoft Org, this repo will contain multiple projects (one for each language)
- add this repo as a submodule to kiota's repository

Someone PRing kiota to change the generation result should:
1. run the generation on the sample repo
2. create a PR in the generation repo
3. update the submodule reference
4. create the PR in Kiota, linking to that generation PR

This has the advantage of:
- providing a generation diff for reviewers
- not polluting the current repo
- force us to start building proper unit tests
- being simple for anyone who knows about submodules
- not being blocking for anyone who doesn't know about submodule

Thoughts?

Reply:
works for me.

Reply:
perfect! I have captured #99 from my previous comment. I'll probably setup the infrastructure in a separate PR based of main and then work my way up on each pending PR with the incremental changes.

Reply:
resolving this comment as you'll see the latest commits come with a submodule and a generation diff. Happy reviews :)